lib/core/src/sdk.rs

```diff
@@ -2164,136 +2196,181 @@ impl LiquidSdk {
+                        &swap_id,
+                        to_state,
+                        server_lockup_tx_id.as_deref(),
+                        user_lockup_tx_id.as_deref(),
+                        claim_tx_id.as_deref(),
+                        refund_tx_id.as_deref(),
+                    )
+                    .await;
+            }
+        }
+
+        // Loop over the recovered chain data for monitored chain send swaps
+        for (swap_id, chain_send_data) in recovered_onchain_data.chain_send {
+            if let Some(to_state) = chain_send_data.derive_partial_state() {
+                let server_lockup_tx_id = chain_send_data
+                    .btc_server_lockup_tx_id
+                    .map(|h| h.txid.to_string());
+                let user_lockup_tx_id = chain_send_data
+                    .lbtc_user_lockup_tx_id
+                    .clone()
+                    .map(|h| h.txid.to_string());
+                let claim_tx_id = chain_send_data.btc_claim_tx_id.map(|h| h.txid.to_string());
+                let refund_tx_id = chain_send_data
+                    .lbtc_refund_tx_id
+                    .clone()
+                    .map(|h| h.txid.to_string());
+                let history_updates = vec![
+                    chain_send_data.lbtc_user_lockup_tx_id,
+                    chain_send_data.lbtc_refund_tx_id,
+                ]
+                .into_iter()
+                .flatten()
+                .collect::<Vec<HistoryTxId>>();
+                for history in history_updates {
+                    if let Some(tx) = tx_map.remove(&history.txid) {
+                        self.persister
+                            .insert_or_update_payment_with_wallet_tx(&tx)?;
+                    }
                 }
-            } else {
-                // Payments that are not directly associated with a swap
-                match payments_before_sync.get(&tx_id) {
-                    None => {
-                        // A completely new payment brought in by this sync, in mempool or confirmed
-                        // Covers events:
-                        // - onchain Receive Pending and Complete
-                        // - onchain Send Complete
+                _ = self
+                    .chain_swap_handler
+                    .update_swap_info(
+                        &swap_id,
+                        to_state,
+                        server_lockup_tx_id.as_deref(),
+                        user_lockup_tx_id.as_deref(),
+                        claim_tx_id.as_deref(),
+                        refund_tx_id.as_deref(),
+                    )
+                    .await;
+            }
+        }
+
+        let payments = self
+            .persister
+            .get_payments_by_tx_id(&ListPaymentsRequest::default())?;
+
+        for tx in tx_map.values() {
+            let tx_id = tx.txid.to_string();
+            let maybe_payment = payments.get(&tx_id);
+            match maybe_payment {
```

Comment:
It's emitted during the claim:
https://github.com/breez/breez-sdk-liquid/pull/564/files#diff-0877f6ddcd2129da682b77db7c9f6ac7247c49fc4e0fbe44756654886eca6cbfL823-R796

The reason being we are already setting the `claim_tx_id` there to avoid a race to claim, but doing so prevents `update_swap_info()` from emitting the event when called using the recovery data (no event is emitted because the swap data struct doesn't change):
https://github.com/breez/breez-sdk-liquid/pull/564/files#diff-f09a3f89b9a7fd7422e34d3a519a8ab878445ab33debc6bc60870fd348347a90R2292-R2302